cmd/svcat/parameters/parameters.go

```diff
@@ -69,33 +69,31 @@ func ParseVariableAssignments(params []string) (map[string]interface{}, error) {
 		}
 
 		storedValue, ok := variables[variable]
-		// Logic to add new value to map variables:
-		// if variable DNE and not exploding params -> variable:value
-		// if variable DNE and exploding params -> variable: map[somekey:somevalue]
-		// Logic to add value to variables where variable already exists in variables:
-		// if variable:value -> variable:[old value, newvalue]
-		// if variable:[value1, value2] -> variable:[value1, value2, newvalue]
-
 		if !ok {
 			if len(dotKeys) == 0 {
 				variables[variable] = value
 			} else {
-				for i := len(dotKeys[1:]); i > 0; i-- {
-					if i == len(dotKeys[1:]) {
-						variables[variable] = map[string]interface{}{dotKeys[i]: value}
-					} else {
-						variables[variable] = map[string]interface{}{dotKeys[i]: variables[variable]}
-					}
+				// evaluate dot params and build map
+				// last element in dot params will have value "value", the rest of map nested
+				// using remaining elements
+				// Ex: A.B.C = Z --> map[A:map[B:map[C:Z]]]
+				variables[variable] = map[string]interface{}{dotKeys[len(dotKeys)-1]: value}
+				for i := len(dotKeys) - 2; i > 0; i-- {
+					variables[variable] = map[string]interface{}{dotKeys[i]: variables[variable]}
 				}
 			}
 		} else {
+			// if variable:value -> variable:[old value, newvalue]
```

Comment:
nit: It would help if these comments were placed immediately before the case statement, instead of grouping them at the top

Reply:
I'll get this into my next update!